src/XMonad/StackSet.hs

```diff
@@ -178,6 +180,23 @@ data Stack a = Stack { focus  :: !a        -- focused thing in this set
     deriving (Show, Read, Eq)
 
 
+instance Functor Stack where
+    fmap f s = Stack
+        (f (focus s))
+        (fmap f (up s))
+        (fmap f (down s))
```

Comment:
This looks `derive`able

Reply:
I didn't want to add DeriveFunctor to the module, but if that's OK then I 100% agree.

Reply:
I think if something can be derived, it should be; I'm sure the xmonad source code can take one more language extension ;)